variable.go

```diff
@@ -18,74 +18,66 @@ import (
 	"sync"
 
 	"github.com/google/cel-go/interpreter"
+	"google.golang.org/protobuf/reflect/protoreflect"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 // variable implements interpreter.Activation, providing a lightweight named
 // variable to cel.Program executions.
-type variable struct {
-	// Next is the parent activation
-	Next interpreter.Activation
-	// Name is the variable's name
-	Name string
-	// Val is the value for this variable
-	Val any
+type variables struct {
+	This  optional[any]
```

Comment:
`this` is not available for partial evaluation. To differentiate between unset and not available, we use an optional type.